src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/SaveTempDataFilter.cs

```diff
@@ -29,16 +29,6 @@ public SaveTempDataFilter(ITempDataDictionaryFactory factory)
         /// <inheritdoc />
         public void OnResourceExecuting(ResourceExecutingContext context)
         {
```

Comment:
Have you tried an end to end test with this? The state that is captured here wouldn't have the action result as its too early in the request execution pipeline. Let's discuss offline.